src/main/java/seedu/address/logic/commands/DeleteShownCommand.java

```diff
@@ -0,0 +1,120 @@
+        checkForEmptyList(lastShownList);
+
+        ensureNotAllAreDeleted(model, lastShownList);
+
+        int deletedCount = getFilteredSize(lastShownList);
+        int leftOver = getLeftOverSize(model, lastShownList);
+
+        deletePersons(model, lastShownList);
+
+        showRemainingPersons(model);
+
+        return new CommandResult(String.format(MESSAGE_SUCCESS,
+                deletedCount, leftOver));
+    }
+
+    /**
+     * Checks if the list is empty.
+     * @param list the list
+     * @throws CommandException if the list is empty
+     */
+    private static void checkForEmptyList(ObservableList<Person> list) throws CommandException {
+        if (list.isEmpty()) {
+            throw new CommandException(MESSAGE_NO_PERSONS);
+        }
+    }
+
+    /**
+     * Shows the remaining persons in the address book.
+     * @param model the model
+     */
+    private static void showRemainingPersons(Model model) {
+        model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
+    }
+
+    /**
+     * Deletes all persons in the last shown list.
+     * @param model the model
+     * @param lastShownList the last shown list
+     */
+    private static void deletePersons(Model model, ObservableList<Person> lastShownList) {
+        for (int i = lastShownList.size() - 1; i >= 0; i--) {
+            model.deletePerson(lastShownList.get(i));
+        }
+    }
+
+    /**
+     * Returns the number of persons in the filtered list.
+     * @param lastShownList the last shown list
+     * @return the number of persons in the filtered list
+     */
+    private static int getFilteredSize(ObservableList<Person> lastShownList) {
+        return lastShownList.size();
+    }
+
+    /**
+     * Returns the total number of persons in the address book.
+     * @param model the model
+     * @return the total number of persons in the address book
+     */
+    private static int getTotalSize(Model model) {
+        AddressBook addressBook = new AddressBook(model.getAddressBook());
+
+        return addressBook.getPersonList().size();
+    }
+
+    /**
+     * Returns the number of persons that will be left over after deletion.
+     * @param model the model
+     * @param lastShownList the last shown list
+     * @return the number of persons that will be left over after deletion
+     */
+    private static int getLeftOverSize(Model model, ObservableList<Person> lastShownList) {
+        return getTotalSize(model) - getFilteredSize(lastShownList);
+    }
+
+    /**
+     * Ensures that not all persons are deleted.
+     * @param model the model
+     * @param lastShownList the last shown list
+     * @throws CommandException if all persons are at risk of being deleted
+     */
+    private static void ensureNotAllAreDeleted(Model model, ObservableList<Person> lastShownList)
+            throws CommandException {
+        if (getFilteredSize(lastShownList) == getTotalSize(model)) {
```

Comment:
Great catch that `deleteshown` should be used with a filter. If user intends to delete all he\she should use the `clear` command